cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -146,6 +143,71 @@ func WriteStaticPodManifests(cfg *kubeadmapi.MasterConfiguration) error {
 	return nil
 }
 
+// getHostPathVolumesForTheControlPlane gets the volumes and volumemounts needed for each control plane component
+func getHostPathVolumesForTheControlPlane(cfg *kubeadmapi.MasterConfiguration) (map[string][]v1.Volume, map[string][]v1.VolumeMount) {
```

Comment:
I'll try to make this a little easier to read then, that was part of my goal. I'll add a struct for it